src/backend/distributed/utils/maintenanced.c

```diff
@@ -683,6 +689,101 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 			timeout = Min(timeout, (StatStatementsPurgeInterval * 1000));
 		}
 
+		pid_t backgroundTaskQueueWorkerPid = 0;
+		BgwHandleStatus backgroundTaskQueueWorkerStatus =
+			backgroundTasksQueueBgwHandle != NULL ? GetBackgroundWorkerPid(
+				backgroundTasksQueueBgwHandle, &backgroundTaskQueueWorkerPid) :
+			BGWH_STOPPED;
+		if (TimestampDifferenceExceeds(lastBackgroundTaskQueueCheck,
+									   GetCurrentTimestamp(),
+									   BackgroundTaskQueueCheckInterval) &&
+			backgroundTaskQueueWorkerStatus == BGWH_STOPPED)
+		{
+			/* clear old background worker for task queue before checking for new tasks */
+			if (backgroundTasksQueueBgwHandle)
+			{
+				TerminateBackgroundWorker(backgroundTasksQueueBgwHandle);
+				pfree(backgroundTasksQueueBgwHandle);
+				backgroundTasksQueueBgwHandle = NULL;
+			}
+
+			StartTransactionCommand();
+
+			bool shouldStartBackgroundTaskQueueBackgroundWorker = false;
+			if (!LockCitusExtension())
+			{
+				ereport(DEBUG1, (errmsg("could not lock the citus extension, "
+										"skipping stat statements purging")));
+			}
+			else if (CheckCitusVersion(DEBUG1) && CitusHasBeenLoaded())
+			{
+				/* perform catalog precheck */
+				shouldStartBackgroundTaskQueueBackgroundWorker =
+					HasRunnableBackgroundTask();
+			}
+
+			CommitTransactionCommand();
+
+			if (shouldStartBackgroundTaskQueueBackgroundWorker)
+			{
+				/*
+				 * Before we start the background worker we want to check if an orphaned
+				 * one is still running. This could happen when the maintenance daemon
+				 * restrated in a way where the background task queue monitor wasn't
+				 * restarted.
+				 *
+				 * To check if an orphaned background task queue monitor is still running
+				 * we quickly acquire the lock without waiting. If we can't acquire the
+				 * lock this was we know that some other backed still has the lock. We
+				 * prevent a new backend from starting and log a warning that we found
+				 * that another process still holds the lock.
+				 */
+				LOCKTAG tag = { 0 };
+				SET_LOCKTAG_CITUS_OPERATION(tag, CITUS_BACKGROUND_TASK_MONITOR);
+				const bool sessionLock = false;
+				const bool dontWait = true;
+				LockAcquireResult locked =
+					LockAcquire(&tag, AccessExclusiveLock, sessionLock, dontWait);
+
+				if (locked == LOCKACQUIRE_NOT_AVAIL)
+				{
+					ereport(WARNING, (errmsg(
```

Comment:
I feel like we could end up warning pretty often after a maintenance daemon restart. Might be nicer to only warn once until we can acquire the lock.